### PR TITLE
Wrapping next AppContext methods

### PR DESCRIPTION
**Pull request by jguz-pubnub**

### CODE REVIEW

File: pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/uuid/GetAllUUIDMetadata.ios.kt [hunk 1]
```diff
@@ -54,27 +54,29 @@ class GetAllUUIDMetadataImpl(
...
-                            eTag = metadata.eTag(),
-                            type = metadata.type(),
-                            status = metadata.status()
-                        )
-                    },
-                    totalCount = it.totalCount()?.intValue ?: 0,
-                    next = it.next()?.end()?.let { hash -> PNPage.PNNext(pageHash = hash) },
-                    prev = it.next()?.start()?.let { hash -> PNPage.PNPrev(pageHash = hash) }
-                )
+            onSuccess = { data, totalCount, next ->
```
Comment (jguz-pubnub) [inline pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/uuid/GetAllUUIDMetadata.ios.kt:57]: I removed Objective-C class like `PubNubGetChannelMetadataResultObjC` and moved all parameters in the closure instead. Is it a better approach or not?

File: pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/channel/GetAllChannelMetadata.ios.kt [hunk 2]
```diff
@@ -51,25 +50,27 @@ class GetAllChannelMetadataImpl(
...
-                            eTag = metadata.eTag(),
-                            type = metadata.type(),
-                            status = metadata.status()
-                        )
-                    },
-                    totalCount = it.totalCount()?.intValue ?: 0,
-                    next = it.next()?.end()?.let { hash -> PNPage.PNNext(pageHash = hash) },
-                    prev = it.next()?.start()?.let { hash -> PNPage.PNPrev(pageHash = hash) }
-                )
+            onSuccess = { data, totalCount, next ->
```
Comment (jguz-pubnub) [inline pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/channel/GetAllChannelMetadata.ios.kt:53]: The same, I also removed Objective-C class like `PubNubGetChannelMetadataResultObjC` and moved all parameters in the closure instead. Is it a better approach or not?

File: pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/channel/SetChannelMetadata.ios.kt [hunk 3]
```diff
@@ -1,9 +1,59 @@
...
+    private val includeCustom: Boolean,
+    private val type: String?,
+    private val status: String?
+): SetChannelMetadata {
+    override fun async(callback: Consumer<Result<PNChannelMetadataResult>>) {
+        pubnub.setChannelMetadataWithChannel(
+            channel = channel,
+            name = name,
+            description = description,
+            custom = null, // TODO: How to retrieve value from CustomObject?
```
Comment (jguz-pubnub) [inline pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/channel/SetChannelMetadata.ios.kt:37]: What should our approach be in this case? I'd like to pass a value from `CustomObject` but it's private for now

Comment (wkal-pubnub) [reply pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/channel/SetChannelMetadata.ios.kt:37]: make CustomObject.value public in iOS for now and pass it here, handling customobjects might change in the future but for now it'll do

File: pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/uuid/GetAllUUIDMetadata.ios.kt (hunk 1 above)
Comment (wkal-pubnub) [reply pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/uuid/GetAllUUIDMetadata.ios.kt:57]: if it makes the code simpler (and maybe smaller - one less class?) then sure, it's not to be used by anyone except us internally, so whatever you prefer here

Comment (jguz-pubnub) [reply pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/uuid/GetAllUUIDMetadata.ios.kt]: I introduced a method like `onSuccessHandler3` we talked about today.
Let me know if you prefer a better name for that

File: pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/channel/SetChannelMetadata.ios.kt (hunk 3 above)
Comment (jguz-pubnub) [reply pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/channel/SetChannelMetadata.ios.kt]: Done

File: pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/channel/GetAllChannelMetadata.ios.kt (hunk 2 above)
Comment (jguz-pubnub) [reply pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/api/endpoints/objects/channel/GetAllChannelMetadata.ios.kt]: Done. I did the same approach that was mentioned previously [here](https://github.com/pubnub/kotlin/pull/229#discussion_r1633431434)

**Review by wkal-pubnub — APPROVED** [review]

**Pull request merged**